dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/GenericAuthorizationFeatureIT.java

```diff
@@ -0,0 +1,1709 @@
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").exists());
+
+        // Verify community A admin has this feature on item 1
+        getClient(communityAAdminToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").exists());
+
+        // Verify collection X admin has this feature on item 1
+        getClient(collectionXAdminToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").exists());
+
+        // Verify item 1 admin doesn’t have this feature on item 1
+        getClient(item1AdminToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").doesNotExist());
+
+        // Verify community A admin doesn’t have this feature on item 2
+        getClient(communityAAdminToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item2.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='"
+                + feature + "')]").doesNotExist());
+
+
+        // grant item 1 admin REMOVE permissions on the item’s owning collection
+        // verify item 1 admin has this feature on item 1
+        context.turnOffAuthorisationSystem();
+        ResourcePolicy removePermission = ResourcePolicyBuilder.createResourcePolicy(context)
+            .withDspaceObject(collectionX)
+            .withAction(Constants.REMOVE)
+            .withUser(item1Writer)
+            .build();
+        context.restoreAuthSystemState();
+
+        // verify item 1 write has this feature on item 1
+        getClient(item1WriterToken).perform(
+            get("/api/authz/authorizations/search/object?embed=feature&uri="
+                + "http://localhost/api/core/items/" + item1.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$._embedded.authorizations[?(@._embedded.feature.id=='canMove')]")
+                .exists());
+
+        context.turnOffAuthorisationSystem();
+        ResourcePolicyBuilder.delete(removePermission.getID());
+        context.restoreAuthSystemState();
```

Comment:
I don't think you need these lines. Since you created `removePermission` from ResourcePolicyBuilder, it should be cleaned up automatically.

Reply:
This has been adjusted